unstructured/ingest/connector/jira.py

```diff
@@ -0,0 +1,486 @@
+            else:
+                logger.error(f"Error: {error} for issue {self.file_meta.issue_key}")
+
+    @property
+    def date_created(self) -> Optional[str]:
+        if not hasattr(self, "metadata_fields"):
+            self.get_metadata_fields()
+
+        return self.metadata_fields["date_created"] if hasattr(self, "metadata_fields") else None
+
+    @property
+    def date_modified(self) -> Optional[str]:
+        if not hasattr(self, "metadata_fields"):
+            self.get_metadata_fields()
+
+        return self.metadata_fields["date_modified"] if hasattr(self, "metadata_fields") else None
+
+    @property
+    def date_processed(self) -> Optional[str]:
+        if not hasattr(self, "metadata_fields"):
+            self.get_metadata_fields()
+
+        return self.metadata_fields["date_processed"] if hasattr(self, "metadata_fields") else None
+
+    @property
+    def exists(self) -> Optional[bool]:
+        """Whether the document exists on the remote source."""
+        if self.check_exists:
+            return self.file_exists
+
+        self.get_metadata_fields()
+
+        return self.file_exists
+
+    def grouping_folder_name(self):
+        if self.file_meta.board_id:
+            return self.file_meta.board_id
+        else:
+            return self.file_meta.project_id
+
+    @property
+    def filename(self):
+        download_file = f"{self.file_meta.issue_id}.txt"
+
+        return (
+            Path(self.standard_config.download_dir) / self.grouping_folder_name() / download_file
+        ).resolve()
+
+    @property
+    def _output_filename(self):
+        """Create output file path."""
+        output_file = f"{self.file_meta.issue_id}.json"
+
+        return (
+            Path(self.standard_config.output_dir) / self.grouping_folder_name() / output_file
+        ).resolve()
+
+    @requires_dependencies(["atlassian"], extras="jira")
+    @BaseIngestDoc.skip_if_file_exists
+    def get_file(self):
+        logger.debug(f"Fetching {self} - PID: {os.getpid()}")
+
+        try:
+            self.jira = self.config.create_session_handle().service
+
+            # GET issue data
+            issue = self.jira.issue(self.file_meta.issue_key)
+            self.check_exists = True
+            self.file_exists = True
+
+            # parse issue data
+            self.document = get_fields_for_issue(issue)
+
+            self.filename.parent.mkdir(parents=True, exist_ok=True)
+
+            with open(self.filename, "w", encoding="utf8") as f:
+                f.write(self.document)
+
+        except HTTPError as error:
+            if error.response.reason == "Not Found":
+                self.check_exists = True
+                self.file_exists = False
+            else:
+                logger.error(f"Error: {error} for issue {self.file_meta.issue_key}")
+
+
+@requires_dependencies(["atlassian"], extras="jira")
+@dataclass
+class JiraConnector(ConnectorCleanupMixin, BaseConnector):
```

Comment:
add data source properties for this? 

i.e. data_created, date_modified, version, record_locator, and source_url?